kuma/banners/templatetags/jinja_helpers.py

```diff
@@ -0,0 +1,11 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+from django_jinja import library
+
+from ..models import Banner
+
+
+@library.global_function
+def get_banners():
+    return Banner.objects.all()
```

Comment:
I appreciate that I'm very very new to the kuma code base so take me with a handful of salt. But, this seems like a not-ideal pattern. Jinja helpers shouldn't be used for get more data. They should be to express/repackage stuff that can't easily be done in template language. 

This function is very powerless. It doesn't support filtering or ordering. 
In particular, the query should probably be: 
```python
Banner.objects.filter(active=True).order_by('priority')
```

A better place to gather the data before template rendering even starts is the view functions. But since, I think, this data isn't related to a specific view, but rather, to *every* view, it should live in its own context processor.

Reply:
So, the template tag will call the function on the context_processor or, should the templates simply use the variable set into context directly?

I ask because I have seen both being used.

Reply:
As far as I can remember, the way it works is something like this:

1. View Python function code does stuff and calls something like `return render(request, 'foo.html', mycontext)`
2. Context processors are looped over and for each one it gets a chance to update that `mycontext` with extra stuff. 
3. The Jinja template is rendered with the fat `mycontext`.

Generally, context processors are really useful for data stuff needed by the `base.html` template. Otherwise, if it depends on some context data, you'd have to explicitly ask every single view function to provide it.

Reply:
That sounds good to me. We could then definitely replace the jinja helper with a context processor, and have it stick the list of banners into the context. @jwhitlock Thoughts?

Reply:
I agree, a context processor is probably the right tech for a site-wide banner.